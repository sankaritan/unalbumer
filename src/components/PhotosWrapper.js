import React, { Component } from "react";
import { connect } from "react-redux";
import {
  getAllPhotosInAlbumsAction,
  getAllPhotosAction,
  createNewAlbumAction
} from "../actions/photosActions";
import { filterUnorganizedPhotos } from "../selectors/photosSelector";
import PropTypes from "prop-types";

export class PhotosWrapper extends Component {
  dataError = null;

  componentDidMount() {
    this.props
      .dispatch(getAllPhotosInAlbumsAction(this.props.oauthToken))
      .catch((error) => {
        this.dataError = error.message;
      });

    this.props
      .dispatch(getAllPhotosAction(this.props.oauthToken))
      .catch((error) => {
        this.dataError = error.message;
      });
  }

  getUnorganizedPhotos() {
    const photos = this.props.unorganizedPhotos;
    return photos.map((photo) => (
      <img className="photo-card" key={photo.id} src={photo.baseUrl} />
    ));
  }

  addToNewAlbum(title) {
    this.props
      .dispatch(
        createNewAlbumAction(
          this.props.oauthToken,
          this.props.unorganizedPhotos,
          title
        )
      )
      .catch((error) => {
        this.dataError = error.message;
      });
  }

  render() {
    return (
      <div>
        {this.props.loggedIn && !this.dataError ? (
          <div>
            <h3>All Unorganized Photos</h3>
            <button onClick={() => this.addToNewAlbum("Unalbumer 1")}>
              Create new album
            </button>
            {this.props.newAlbum && (
              <p>
                New empty album{" "}
                <a href={this.props.newAlbum.productUrl}>
                  <b>{this.props.newAlbum.title}</b>
                </a>{" "}
                was created.
              </p>
            )}
            <div>{this.getUnorganizedPhotos()}</div>
          </div>
        ) : (
          <div>
            <p>
              You seem to be logged out or you failed to fetch data from Google!
            </p>
          </div>
        )}
      </div>
    );
  }
}

PhotosWrapper.propTypes = {
  dispatch: PropTypes.func,
  oauthToken: PropTypes.string,
  loggedIn: PropTypes.bool,
  photosInAlbums: PropTypes.array,
  allPhotos: PropTypes.array,
  unorganizedPhotos: PropTypes.array,
  newAlbum: PropTypes.object
};

const mapStateToProps = (state) => ({
  oauthToken: state.user.oauthToken,
  loggedIn: state.user.loggedIn,
  photosInAlbums: state.photos.photosInAlbums,
  allPhotos: state.photos.allPhotos,
  unorganizedPhotos: filterUnorganizedPhotos(
    state.photos.allPhotos,
    state.photos.photosInAlbums
  ),
  newAlbum: state.photos.newAlbum
});

export default connect(mapStateToProps)(PhotosWrapper);
