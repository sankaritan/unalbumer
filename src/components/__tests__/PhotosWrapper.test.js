import React from "react";
import { PhotosWrapper } from "../PhotosWrapper";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("Page Header component", () => {
  const mockDispatch = () => {
    return new Promise(() => {});
  };

  const testOptions = [
    {
      dispatch: mockDispatch,
      oauthToken: "token",
      loggedIn: true,
      photosInAlbums: [1, 2, 3],
      allPhotos: [1, 2, 3, 4, 5],
      unorganizedPhotos: [
        { id: 4, filename: "image4.png", baseUrl: "www.google.com" },
        { id: 5, filename: "image5.png", baseUrl: "www.google.com" }
      ],
      newAlbum: null
    },
    {
      dispatch: mockDispatch,
      oauthToken: "token",
      loggedIn: false,
      photosInAlbums: [1, 2, 3],
      allPhotos: [1, 2, 3, 4, 5],
      unorganizedPhotos: [
        { id: 4, filename: "image4.png", baseUrl: "www.google.com" },
        { id: 5, filename: "image5.png", baseUrl: "www.google.com" }
      ],
      newAlbum: null
    },
    {
      dispatch: mockDispatch,
      oauthToken: "token",
      loggedIn: true,
      photosInAlbums: [1, 2, 3],
      allPhotos: [1, 2, 3, 4, 5],
      unorganizedPhotos: [],
      newAlbum: { title: "title", productUrl: "https://www.album68.com/album" }
    }
  ];

  testOptions.forEach((props) => {
    it("PhotosWrapper renders correctly", () => {
      const component = shallow(<PhotosWrapper {...props} />).debug();
      expect(component).toMatchSnapshot();
    });
  });
});
