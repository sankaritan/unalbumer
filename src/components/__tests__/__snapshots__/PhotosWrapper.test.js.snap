// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header component PhotosWrapper renders correctly 1`] = `
"<div>
  <div>
    <h3>
      All Unorganized Photos
    </h3>
    <button id=\\"new-album-button\\" onClick={[Function: onClick]}>
      Create new album
    </button>
    <div>
      <img className=\\"photo-card\\" alt=\\"image4.png\\" src=\\"www.google.com\\" />
      <img className=\\"photo-card\\" alt=\\"image5.png\\" src=\\"www.google.com\\" />
    </div>
  </div>
</div>"
`;

exports[`Page Header component PhotosWrapper renders correctly 2`] = `
"<div>
  <div>
    <p>
      You seem to be logged out or you failed to fetch data from Google!
    </p>
  </div>
</div>"
`;

exports[`Page Header component PhotosWrapper renders correctly 3`] = `
"<div>
  <div>
    <h3>
      All Unorganized Photos
    </h3>
    <button id=\\"new-album-button\\" onClick={[Function: onClick]}>
      Create new album
    </button>
    <p id=\\"new-album-link\\">
      New empty album
       
      <a href=\\"https://www.album68.com/album\\">
        <b>
          title
        </b>
      </a>
       
      was created.
    </p>
    <div />
  </div>
</div>"
`;
